From 02c32d3dd5e8a4349d8b39e0f1bb330ac663cda4 Mon Sep 17 00:00:00 2001
From: Michal Simek <monstr@monstr.eu>
Date: Thu, 10 May 2012 10:26:12 +0200
Subject: [PATCH 1003/1566] remoteproc: zynq: Turn on SMP mode when probing
 failed.

commit  1ded255562963624fe0fb3e216497eff5ed3c9e4 from
https://github.com/Xilinx/linux-xlnx.git

When probing failed, Linux can return to SMP mode again.
Plus correct email address and add module author.

Signed-off-by: Michal Simek <monstr@monstr.eu>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/remoteproc/zynq_remoteproc.c |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/drivers/remoteproc/zynq_remoteproc.c b/drivers/remoteproc/zynq_remoteproc.c
index d40e65c..108d2ab 100644
--- a/drivers/remoteproc/zynq_remoteproc.c
+++ b/drivers/remoteproc/zynq_remoteproc.c
@@ -1,7 +1,7 @@
 /*
  * Zynq Remote Processor driver
  *
- * Copyright (C) 2012 Michal Simek <monstr@monst.eu>
+ * Copyright (C) 2012 Michal Simek <monstr@monstr.eu>
  * Copyright (C) 2012 PetaLogix
  *
  * Based on origin OMAP Remote Processor driver
@@ -34,6 +34,7 @@
 #include <asm/cacheflush.h>
 #include <mach/system.h>
 #include <linux/slab.h>
+#include <linux/cpu.h>
 
 #include "remoteproc_internal.h"
 
@@ -279,6 +280,11 @@ ipi_fault:
 irq_fault:
 	clear_irq(pdev);
 
+	/* Cpu can't be power on - for example in nosmp mode */
+	ret = cpu_up(1);
+	if (ret)
+		dev_err(&pdev->dev, "Can't power on cpu1 %d\n", ret);
+
 	return ret;
 }
 
@@ -320,5 +326,6 @@ module_platform_driver(zynq_remoteproc_driver);
 module_param(firmware, charp, 0);
 MODULE_PARM_DESC(firmware, "Override the firmware image name. Default value in DTS.");
 
+MODULE_AUTHOR("Michal Simek <monstr@monstr.eu");
 MODULE_LICENSE("GPL v2");
 MODULE_DESCRIPTION("Zynq remote processor control driver");
-- 
1.7.5.4

